cmd/user-cluster-controller-manager/main.go

```diff
@@ -382,7 +382,7 @@ func main() {
 		log.Info("Registered constraintsyncer controller")
 	}
 
-	if err := applicationinstallationcontroller.Add(rootCtx, log, seedMgr, mgr, isPausedChecker, &applications.ApplicationManager{}); err != nil {
+	if err := applicationinstallationcontroller.Add(rootCtx, log, seedMgr, mgr, isPausedChecker, &applications.ApplicationManager{SecretNamespace: runOp.namespace}); err != nil {
```

Comment:
As you know, application credentials are replicated from master to seed clusters into "kubermatic" namespace by the `application-secret-synchronizer`. 

The user cluster does not have the permission to read secrets into the "kubermatic" namespace.  I could add RBAC to provide access to user-cluster-controller-manager to the kubermatic namespace. Nevertheless, it feels wrong as this namespace has many sensible secrets (especially on master/seed setup). 

After discussing with @sig-cluster-management, we agree the better solution is to have a new controller in the seed-controller-manager that replicates application secrets into all cluster namespaces (e.g cluster-znx6n7jbdz). (it will be implemented in another PR...)

`runOp.Namespace`is the namespace where the user-cluster-controller is running (ie cluster-<cluster_id>).  Currently for testing, I manually copy application credentials into the cluster namespace....

Reply:
ah okay understood.